src/modules/hip/kernel/box_filter.hpp

```diff
@@ -0,0 +1,775 @@
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack1(src);
+    dst_f8->x.y = fmaf(src_f, 0.01234568f, dst_f8->x.y);
+    dst_f8->x.z = fmaf(src_f, 0.01234568f, dst_f8->x.z);
+    dst_f8->x.w = fmaf(src_f, 0.01234568f, dst_f8->x.w);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack2(src);
+    dst_f8->x.z = fmaf(src_f, 0.01234568f, dst_f8->x.z);
+    dst_f8->x.w = fmaf(src_f, 0.01234568f, dst_f8->x.w);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack3(src);
+    dst_f8->x.w = fmaf(src_f, 0.01234568f, dst_f8->x.w);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src = src_uchar4[3];
+    src_f = rpp_hip_unpack0(src);
+    dst_f8->y.x = fmaf(src_f, 0.01234568f, dst_f8->y.x);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack1(src);
+    dst_f8->y.y = fmaf(src_f, 0.01234568f, dst_f8->y.y);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack2(src);
+    dst_f8->y.z = fmaf(src_f, 0.01234568f, dst_f8->y.z);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+    src_f = rpp_hip_unpack3(src);
+    dst_f8->y.w = fmaf(src_f, 0.01234568f, dst_f8->y.w);
+}
+
+// Handles PKD3->PKD3 for any combination of kernelSize = 3/5/7/9 and T = U8/F32/F16/I8
+template <typename T>
+__global__ void box_filter_pkd_tensor(T *srcPtr,
+                                      uint2 srcStridesNH,
+                                      T *dstPtr,
+                                      uint2 dstStridesNH,
+                                      uint kernelSize,
+                                      uint padLength,
+                                      uint2 tileSize,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+    int hipThreadIdx_x8 = hipThreadIdx_x << 3;
+    int id_x_o = (hipBlockIdx_x * tileSize.x * 8) + hipThreadIdx_x8;
+    int id_y_o = hipBlockIdx_y * tileSize.y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    int id_x_i = id_x_o - padLength;
+    int id_y_i = id_y_o - padLength;
+    d_float24 sum_f24;
+    __shared__ uchar src_lds[48][128];
```

Comment:
why do you need to define a two-dimensional shared memory? can't you just use shared uchar src_lds[6144]? and use the stride to access the shared memory?

Reply:
I saw no perf difference in using single array, and secondly the addressing of the tiles for the R/G/B locations in lds became easier with 2D.